testing/src/test/java/io/stargate/it/cql/BoundStatementTest.java

```diff
@@ -0,0 +1,172 @@
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.QueryTrace;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.data.ByteUtils;
+import com.datastax.oss.driver.api.core.servererrors.InvalidQueryException;
+import com.datastax.oss.driver.api.core.servererrors.ProtocolError;
+import io.stargate.it.storage.ClusterConnectionInfo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Note: all tests related to unset values assume that the protocol version is always 4 or higher.
+ */
+public class BoundStatementTest extends JavaDriverTestBase {
+
+  private static final String KEY = "test";
+  private static final int VALUE = 7;
+
+  public BoundStatementTest(ClusterConnectionInfo backend) {
+    super(backend);
+  }
+
+  @BeforeEach
+  public void setupSchema() {
+    // table where every column forms the primary key.
+    session.execute("CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))");
+    for (int i = 0; i < 100; i++) {
+      session.execute(
+          SimpleStatement.builder("INSERT INTO test (k, v) VALUES (?, ?)")
+              .addPositionalValues(KEY, i)
+              .build());
+    }
+
+    // table with simple primary key, single cell.
+    session.execute("CREATE TABLE IF NOT EXISTS test2 (k text primary key, v0 int)");
+
+    // table with composite partition key
+    session.execute(
+        "CREATE TABLE IF NOT EXISTS test3 "
+            + "(pk1 int, pk2 int, v int, "
+            + "PRIMARY KEY ((pk1, pk2)))");
+  }
+
+  @Test
+  public void should_use_positional_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, VALUE));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  public void should_use_named_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (:k, :v)");
+    session.execute(
+        prepared.boundStatementBuilder().setString("k", KEY).setInt("v", VALUE).build());
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  public void should_allow_nulls_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, null));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.isNull("v0")).isTrue();
+  }
+
+  @Test
+  @Disabled("Returns an empty result set instead of failing")
+  public void should_fail_if_missing_values() {
```

Comment:
Codacy found an issue: [The JUnit 5 test method name 'should_fail_if_missing_values' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263918)